src/mime/media_type.cr

```diff
@@ -0,0 +1,491 @@
+require "uri"
+require "http"
+require "mime"
+
+module MIME
+  # A MediaType describes a MIME content type with optional parameters.
+  struct MediaType
+    getter media_type : String
+
+    # Creates a new `MediaType` instance.
+    def initialize(@media_type, @params = {} of String => String)
+    end
+
+    def to_s(io : IO)
+      io << media_type
+
+      @params.each do |key, value|
+        next unless MIME::MediaType.token? key
```

Comment:
Should we really be silently ignoring this?

Reply:
Maybe it should raise when setting such a key?